elastic/kibana
# [Response Ops][Task Manager] Changing task manager `schedule.interval` schema to string from duration

## Summary

Currently, when task manager schema changes are migrated down (which can occur during a release when there are multiple Kibana nodes running and they get updated one after another), we are running into this bug: https://github.com/elastic/kibana/issues/204395 where the task  `schedule.interval` gets mutated from expected values (like `60s`) to unallowed (but still valid for the schema) values (like `PT1M`).

This changes the schema for `schedule.interval` to use a string with validation function.

## To verify
1. Run Kibana on `main` and create some rules that run frequently.
2. "Upgrade" to this PR branch and verify that rules continue to run.
3. "Downgrade" back to `main` and verify that rules continue to run.
